### PR TITLE
Alter the join/leave message style

### PR DESCRIPTION
**Pull request by sarahsturgeon**
This should make the messages look like the rest, but with a small embed saying when they spawned and when they disconnected.

Decided to go with `PlayerInitialSpawn` because people will probably care more about when someone spawns than when they connect. (Also I get access to the actual player object at that point)

### CODE REVIEW

File: web/discord_relay/queue.go [hunk 1]
```diff
@@ -59,8 +54,14 @@ func sendConnectMessage(discord *discordgo.Session, message MessageStruct) {
...
 		},
-		Content:  fmt.Sprintf("%v %v | %v has connected to the server", JOIN_EMOJI, message.Data.SteamName, message.Data.SteamId),
-		Username: "Relay",
+		Username:  message.Data.SteamName,
+		AvatarURL: message.Data.Avatar,
+		Embeds: []discordgo.MessageEmbed{
+		    &discordgo.MessageEmbed{
+                Description: fmt.Sprintf("%v ***Connected to the server***", JOIN_EMOJI),
+                Color:       65280,
+            },
```
Comment (sarahsturgeon) [inline web/discord_relay/queue.go]: This is wrong, I'm not sure how to properly do it; @plally

Comment (plally) [reply web/discord_relay/queue.go]: `0x` before the number if its a hex color

**Comment by wrefgtzweve** [comment]
image
fyi, PlayerDisconnected can return nil on the steamID

**Comment by sarahsturgeon** [comment]
I've never actually seen it do that, personally. I've taken the risk before and haven't been burned yet

Worst case we'll store the data in a table by their Steam ID and then use the game event to look up the info.

File: web/discord_relay/queue.go (hunk 1 above)
Comment (sarahsturgeon) [reply web/discord_relay/queue.go]: The embed structure is also wrong, how do I make an array of one embed?

**Comment by wrefgtzweve** [comment]
Fully agree with the embeds, was talking with pcl about them. Great idea!

File: web/discord_relay/queue.go (hunk 1 above)
Comment (plally) [reply web/discord_relay/queue.go]:
```go
[]*discordgo.MessageEmbed{
    {
      Description: "description",
    }
}
```
im not sure you can have an embed without a title can you?

File: moon/cfc_chat_transit/server/init.moon [hunk 2]
```diff
@@ -79,39 +78,47 @@ ChatTransit.ReceiveMessage = (ply, text, teamChat) =>
 
     @Logger\debug "Sent message '#{text}' to websocket"
 
-ChatTransit.PlayerConnected = (data) =>
-    steamName = data.name
-    steamId = data.networkid
+ChatTransit.PlayerInitialSpawn = (ply) =>
+    sendMessage = (attempts=1) ->
+        if attempts >= 5
+            return @Logger\error "PlayerSummary didn't exist in time to send an on-spawn message"
```
Comment (plally) [inline moon/cfc_chat_transit/server/init.moon]: I think logging and still continuing to send the join alert with the default avatar would be better

Comment (sarahsturgeon) [reply moon/cfc_chat_transit/server/init.moon]: Yeah good call

File: moon/cfc_chat_transit/server/init.moon [hunk 3]
```diff
@@ -79,39 +78,47 @@ ChatTransit.ReceiveMessage = (ply, text, teamChat) =>
 
     @Logger\debug "Sent message '#{text}' to websocket"
 
-ChatTransit.PlayerConnected = (data) =>
-    steamName = data.name
-    steamId = data.networkid
+ChatTransit.PlayerInitialSpawn = (ply) =>
+    sendMessage = (attempts=1) ->
```
Comment (plally) [inline moon/cfc_chat_transit/server/init.moon:82]: where is this initially getting called?

**Comment by wrefgtzweve** [comment]
> I've never actually seen it do that, personally. I've taken the risk before and haven't been burned yet
> 
> Worst case we'll store the data in a table by their Steam ID and then use the game event to look up the info.

Another reason i was using the game events is because they give much more information, like a very neat one to show disconnect reasons which the normal hook cant.

**Review by plally — APPROVED** [review]

**Comment by wrefgtzweve** [comment]
Might want to set the convar value to a default otherwise the entire script will break

**Comment by plally** [comment]
If not that you could leave a note in the readme about setting the realm at least

**Pull request merged**